File: proj.android_studio/ee-x-store/src/main/java/com/ee/store/StoreUtils.java
package com.ee.store;

import androidx.annotation.NonNull;

import com.android.billingclient.api.Purchase;
import com.android.billingclient.api.PurchaseHistoryRecord;
import com.android.billingclient.api.SkuDetails;

import java.util.HashMap;
import java.util.Map;

class StoreUtils {
    @NonNull
    static Map<String, Object> convertSkuDetailsToDictionary(@NonNull SkuDetails item) {
        Map<String, Object> dict = new HashMap<>();
        dict.put("description", item.getDescription());
        dict.put("free_trial_period", item.getFreeTrialPeriod());
        dict.put("price", item.getPrice());
        dict.put("price_amount_micros", item.getPriceAmountMicros());
        dict.put("price_currency_code", item.getPriceCurrencyCode());
        dict.put("sku", item.getSku());
        dict.put("subscription_period", item.getSubscriptionPeriod());
        dict.put("title", item.getTitle());
        return dict;
    }

    @NonNull
    static Map<String, Object> convertPurchaseToDictionary(@NonNull Purchase item) {
        Map<String, Object> dict = new HashMap<>();
        dict.put("package_name", item.getPackageName());
        dict.put("purchase_state", item.getPurchaseState());
        dict.put("purchase_time", item.getPurchaseTime());
        dict.put("purchase_token", item.getPurchaseToken());
        dict.put("signature", item.getSignature());
        dict.put("sku", item.getSku());
        dict.put("is_acknowledged", item.isAcknowledged());
        dict.put("is_auto_renewing", item.isAutoRenewing());
        return dict;
    }

    @NonNull
    static Map<String, Object> convertRecordToDictionary(@NonNull PurchaseHistoryRecord item) {
        Map<String, Object> dict = new HashMap<>();
        dict.put("purchase_time", item.getPurchaseTime());
        dict.put("purchase_token", item.getPurchaseToken());
        dict.put("signature", item.getSignature());
        dict.put("sku", item.getSku());
        return dict;
    }
}

File: proj.android_studio/ee-x-store/src/main/java/com/ee/internal/StoreUtils.kt
package com.ee.internal

import com.android.billingclient.api.Purchase
import com.android.billingclient.api.PurchaseHistoryRecord
import com.android.billingclient.api.SkuDetails

internal object StoreUtils {
    fun convertSkuDetailsToDictionary(item: SkuDetails): Map<String, Any> {
        val dict: MutableMap<String, Any> = HashMap()
        dict["description"] = item.description
        dict["free_trial_period"] = item.freeTrialPeriod
        dict["price"] = item.price
        dict["price_amount_micros"] = item.priceAmountMicros
        dict["price_currency_code"] = item.priceCurrencyCode
        dict["sku"] = item.sku
        dict["subscription_period"] = item.subscriptionPeriod
        dict["title"] = item.title
        return dict
    }

    fun convertPurchaseToDictionary(item: Purchase): Map<String, Any> {
        val dict: MutableMap<String, Any> = HashMap()
        dict["package_name"] = item.packageName
        dict["purchase_state"] = item.purchaseState
        dict["purchase_time"] = item.purchaseTime
        dict["purchase_token"] = item.purchaseToken
        dict["signature"] = item.signature
        dict["sku"] = item.sku
        dict["is_acknowledged"] = item.isAcknowledged
        dict["is_auto_renewing"] = item.isAutoRenewing
        return dict
    }

    fun convertRecordToDictionary(item: PurchaseHistoryRecord): Map<String, Any> {
        val dict: MutableMap<String, Any> = HashMap()
        dict["purchase_time"] = item.purchaseTime
        dict["purchase_token"] = item.purchaseToken
        dict["signature"] = item.signature
        dict["sku"] = item.sku
        return dict
    }
}